jwst/barshadow/bar_shadow.py

```diff
@@ -105,6 +105,8 @@ def do_correction(input_model, barshadow_model):
                 slitlet.err /= correction
                 slitlet.var_poisson /= correction**2
                 slitlet.var_rnoise /= correction**2
+                if slitlet.var_flat is not None and np.size(slitlet.var_flat) > 0:
```

Comment:
Neither of these checks on `slitlet.var_flat` will do anything because as soon as it is accessed, it becomes an array of zeros the same size as `slitlet.data`.  It will never be `None` and will always have a size.

I.e. the above line will always evalutate to `True`.  I don't know if this is desirable.  Perhaps this is due to the default value for `var_flat` in the schema?

This is a larger `datamodels` issue I think.

Reply:
Some is true for all the checks elsewhere.

Reply:
```
In [17]: im = datamodels.SlitModel((2, 10))                                                                                                                              

In [18]: im._asdf.tree                                                                                                                                                   
Out[18]: {'meta': {'date': '2019-05-03T15:24:07.530', 'model_type': 'SlitModel'}}

In [19]: im.data is None                                                                                                                                                 
Out[19]: False

In [20]: im._asdf.tree                                                                                                                                                   
Out[20]: 
{'meta': {'date': '2019-05-03T15:24:07.530', 'model_type': 'SlitModel'},
 'data': array([[0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
        [0., 0., 0., 0., 0., 0., 0., 0., 0., 0.]], dtype=float32)}

In [21]: np.size(im.var_flat) > 0                                                                                                                                        
Out[21]: True

In [22]: im._asdf.tree                                                                                                                                                   
Out[22]: 
{'meta': {'date': '2019-05-03T15:24:07.530', 'model_type': 'SlitModel'},
 'data': array([[0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
        [0., 0., 0., 0., 0., 0., 0., 0., 0., 0.]], dtype=float32),
 'var_flat': array([[0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
        [0., 0., 0., 0., 0., 0., 0., 0., 0., 0.]], dtype=float32)}
```

Reply:
This is why I added those checks:

    dm = datamodels.open("jw87600017001_02101_00002_nis_extract_2d.fits")
    type(dm)
    <class 'jwst.datamodels.multislit.MultiSlitModel'>
    slit = dm.slits[0]
    slit.var_poisson.shape
    (0, 0, 0)
    slit.var_rnoise.shape
    (0, 0, 0)
    np.nanmean(slit.var_poisson)
    __main__:1: RuntimeWarning: Mean of empty slice
    nan

    catfits.py jw87600017001_02101_00002_nis_extract_2d.fits
    HDU#  FILENAME             EXTNAME    EXTVER DIMENS         BITPIX
    0     jw87600017001_02101_00002_nis_extract_2d.fits         8
    1       IMAGE              SCI        1      9x105          -32
    2       IMAGE              DQ         1      9x105          32
    3       IMAGE              ERR        1      9x105          -32
    4       BINTABLE           ASDF              1Fx1R          8